graspologic/cluster/rclust.py

```diff
@@ -0,0 +1,377 @@
+        select the best model based on a certain criterion for each split
+    min_split : int, optional (default=1)
+        The minimum number of samples allowed in a leaf cluster
+    max_level : int, optional (default=20)
+        The maximum level to cluster
+    delta_criter : float or None, positive, default=None
+        The smallest difference between selection criterion values of a new
+        model and the current model that is required to accept the new model
+    likelihood_ratio : float or None, in range (0,1), default=None
+        The significance threshold of p-value for likelihood ratio test
+
+    Attributes
+    ----------
+    results_ : dict
+        Contains information about clustering results on a cluster
+        Items are:
+
+        'model' : GaussianMixture (or KMeans) object if "GMM" (or "KMeans")
+            is selected to perform clustering
+        'criter' : float
+            Bayesian (or Akaike) Information Criterion if "bic" (or "aic")
+            is chosen to select the best model
+        'n_components' : int
+            number of components or clusters
+    predictions : array_like, shape (n_samples, n_levels)
+
+    See Also
+    --------
+    graspy.cluster.AutoGMMCluster
+
+    References
+    ----------
+    .. [1]  Lyzinski, V., Tang, M., Athreya, A., Park, Y., & Priebe, C. E
+            (2016). Community detection and classification in hierarchical
+            stochastic blockmodels. IEEE Transactions on Network Science and
+            Engineering, 4(1), 13-26.
+    """
+
+    def __init__(
+        self,
+        selection_criteria=None,
+        cluster_method="GMM",
+        parent=None,
+        min_components=1,
+        max_components=10,
+        cluster_kws={},
+        min_split=1,
+        max_level=20,
+        delta_criter=None,
+        likelihood_ratio=None,
+    ):
+
+        _check_common_inputs(
+            min_components, max_components, cluster_kws, embed_kws=None
+        )
+
+        if cluster_method not in ["GMM", "KMeans", "Spherical-KMeans"]:
+            msg = "clustering method must be one of"
+            msg += "{GMM, Kmeans, Spherical-KMeans}"
+            raise ValueError(msg)
+
+        if delta_criter is not None:
+            if delta_criter <= 0:
+                raise ValueError("delta_criter must be positive")
+
+        if likelihood_ratio is not None:
+            if likelihood_ratio <= 0 or likelihood_ratio >= 1:
+                raise ValueError("likelihood_ratio must be in (0,1)")
+
+        self.parent = parent
+        self.min_components = min_components
+        self.max_components = max_components
+        self.cluster_method = cluster_method
+        self.min_split = min_split
+        self.selection_criteria = selection_criteria
+        self.cluster_kws = cluster_kws
+        self.min_split = min_split
+        self.max_level = max_level
+        self.delta_criter = delta_criter
+        self.likelihood_ratio = likelihood_ratio
```

Comment:
later in the code I never see you specify a value for these if they were passed as `None`. what happens then?

Reply:
i just added some explanation saying that if None, no such restriction on `delta-criter` applied. Also, after some experimentation, I could not find a proper case to show the effect of setting a significance threshold for likelihood_ratio test on improving the results. It seems that even when "BIC" improves a lot (several hundreds), the log_likehood changes very little so the p-value was really large and I couldn't get it pass the test. I think I'm gonna put a pin on this and do more research and add a working version later.

Reply:
sounds great to me